File: 274/solution.py
from typing import List


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        counter = [0] * (n + 1)
        for c in citations:
            if c >= n:
                counter[n] += 1
            else:
                counter[c] += 1
        res = 0
        for i in range(n, -1, -1):
            res += counter[i]
            if res >= i:
                return i
        return 0

File: 274/Solution.java
class Solution {
    public int hIndex(int[] citations) {
        int n = citations.length;
        int res = 0;
        int[] counter = new int[n + 1];
        for (int i = 0; i < n; i++) {
            if (citations[i] >= n) {
                counter[n]++;
            } else {
                counter[citations[i]]++;
            }
        }
        for (int i = n; i >= 0; i--) {
            res += counter[i];
            if (res >= i) {
                return i;
            }
        }
        return 0;
    }
}